jbmc/src/java_bytecode/README.md

```diff
@@ -39,9 +39,160 @@ To be documented.
 
 To be documented.
 
-\section java-bytecode-remove-exceptions remove_exceptions
+\section java-bytecode-remove-exceptions Remove exceptions
 
-To be documented.
+When `remove_exceptions` is called on the goto model, the goto model contains
+complex instructions such as `CATCH-POP`, `CATCH-PUSH` and `THROW`. In order to
+analyze the goto model, the instructions must be simplified to use more basic
+instructions - this is called "lowering". This class lowers the `CATCH` and
+`THROW` instructions.
+
+Exceptions that have been thrown, but not yet caught are stored the global
```

Comment:
perhaps reword to be clearer 

`THROW` instructions are replaced by assigning to `@inflight_exception` and a `goto` to end of the function. 
`CATCH` instructions are replaced by a check of the `@inflight_exception` and setting it to null

Reply:
If the bit about the goto statement is right? In your below example it doesn't seem to introduce a GOTO so perhaps that is already present.

Reply:
Actually it'd be good to split out `CATCH` into `CATCH-PUSH` `CATCH-POP` and `EXCEPTION LANDING PAD` into roughly what they get translated into

Reply:
It introduces THEN GOTO 4